deltachat-ffi/src/lib.rs

```diff
@@ -4424,3 +4425,77 @@ pub unsafe extern "C" fn dc_accounts_get_next_event(
         .map(|ev| Box::into_raw(Box::new(ev)))
         .unwrap_or_else(ptr::null_mut)
 }
+
+#[cfg(feature = "jsonrpc")]
+mod jsonrpc {
+    use super::*;
+    use deltachat_jsonrpc::api::CommandApi;
+    use deltachat_jsonrpc::yerpc::{MessageHandle, RpcHandle};
+
+    pub struct dc_json_api_instance_t {
+        receiver: async_std::channel::Receiver<deltachat_jsonrpc::yerpc::Message>,
+        handle: MessageHandle<CommandApi>,
+    }
+
+    #[no_mangle]
+    pub unsafe extern "C" fn dc_get_json_api(
+        account_manager: *mut dc_accounts_t,
+    ) -> *mut dc_json_api_instance_t {
+        if account_manager.is_null() {
+            eprintln!("ignoring careless call to dc_get_json_api()");
+            return ptr::null_mut();
+        }
+
+        let cmd_api =
+            deltachat_jsonrpc::api::CommandApi::new_from_cffi((*account_manager).inner.clone());
+
+        let (request_handle, receiver) = RpcHandle::new();
+        let handle = MessageHandle::new(request_handle, cmd_api);
+
+        let instance = dc_json_api_instance_t { receiver, handle };
+
+        Box::into_raw(Box::new(instance))
+    }
+
+    #[no_mangle]
+    pub unsafe extern "C" fn dc_json_api_unref(json_api_instance: *mut dc_json_api_instance_t) {
+        if json_api_instance.is_null() {
+            eprintln!("ignoring careless call to dc_json_api_unref()");
```

Comment:
is there any regular circumstance where this is expected to be called with a null pointer? If not, should we not  panic if called with null pointers instead of ignoring these invalid calls?

Same for the other null pointer checks.

Reply:
this is how we do this stuff in the cffi, feel free to open an general issue about it

Reply:
OK, then let's keep this here as well. I'm not too familiar with the design of the CFFI API, was just wondering.